File: services/image_gen_service.py
import base64
from pathlib import Path
from config import ModelConfig
from services.provider_api import (
    build_edit_payload,
    encode_image_path_as_data_url,
    extract_image_result,
    send_request,
)


class ImageGenService:
    def __init__(self, config: ModelConfig) -> None:
        self.config = config

    def translate_page(
        self,
        image_path: Path,
        prompt: str,
        output_dir: Path,
        output_filename: str,
    ) -> Path:
        data_url = encode_image_path_as_data_url(image_path)
        payload = build_edit_payload(
            model=self.config.model,
            prompt=prompt,
            edit_image_data_url=data_url,
        )
        sse_text = send_request(self.config.base_url, self.config.api_key, payload)
        result = extract_image_result(sse_text)
        image_bytes = base64.b64decode(result["result"])
        output_path = output_dir / output_filename
        output_path.write_bytes(image_bytes)
        return output_path

File: src/main/image-gen-service.ts
import { writeFileSync } from 'fs'
import { join } from 'path'
import { readImageAsBase64, ensureOutputDir } from './file-service'
import { buildEditPayload, sendImageGenRequest, extractImageResult } from './provider-api'
import { resizeImageIfNeeded } from './image-utils'
import { type ModelConfig } from './config'

export async function translatePageImage(
  config: ModelConfig,
  imagePath: string,
  prompt: string,
  outputDir: string,
  outputFilename: string
): Promise<string> {
  const { base64, mimeType } = readImageAsBase64(imagePath)
  const resized = await resizeImageIfNeeded(Buffer.from(base64, 'base64'))
  const dataUrl = `data:${mimeType};base64,${resized.toString('base64')}`

  const payload = buildEditPayload(config.model, prompt, dataUrl)
  const sseText = await sendImageGenRequest(config.base_url, config.api_key, payload)
  const { result } = extractImageResult(sseText)

  ensureOutputDir(outputDir)
  const outputPath = join(outputDir, outputFilename)
  writeFileSync(outputPath, Buffer.from(result, 'base64'))
  return outputPath
}
